apollo-federation/src/query_graph/graph_path.rs

```diff
@@ -2686,8 +2673,11 @@ impl OpGraphPath {
             runtime_types_before_tail_if_last_is_cast: None,
             // TODO: The JS codebase copied this from the current path, which seems like a bug.
             defer_on_tail: self.defer_on_tail.clone(),
-            context_to_selection: self.context_to_selection.clone(),
```

Comment:
Bugs:
- These vectors should be truncated similarly to how `edges` is (really we should consider moving to a vector of structs, but that's a follow-up). Note this bug is also present in JS QP.